README.md

```diff
@@ -2,102 +2,78 @@
-  * Pokes PV queue with name of the detached PV. This triggers removal of finalizer on PV, if needed.
+## Compatibility
 
-* Processes added/updated PV to remove finalizer on PVs:
-  * Ignore PVs that don't have DeletionTimestamp.
-  * Checks that the PV is not used by any `VolumeAttachment` instance.
-  * Removes Attacher's finalizer on the PV if so.
-  * On any error, the PV is re-queued with exponential backoff.
+This information reflects the head of this branch.
 
-#### Concurrency
+| Compatible with CSI Version                                                                | Container Image                     | Min K8s Version |
+| ------------------------------------------------------------------------------------------ | ------------------------------------| --------------- |
+| [CSI Spec v0.3.0](https://github.com/container-storage-interface/spec/releases/tag/v0.3.0) | quay.io/k8scsi/csi-attacher:v0.2.0  | 1.10            |
+| [CSI Spec v0.3.0](https://github.com/container-storage-interface/spec/releases/tag/v0.3.0) | quay.io/k8scsi/csi-attacher:v0.4.2  | 1.12            |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | quay.io/k8scsi/csi-attacher:v1.0.1  | 1.13            |
 
-Both PV queue and `VolumeAttachment` queue run in parallel. To ensure that removal of PV finalizers work without races:
+## Usage
 
-* The controller attaches PVs only when the PV has no DeletionTimestamp and has attacher's finalizer.
-* The controller removes finalizer only from PVs that have DeletionTimestamp.
+It is necessary to create a new service account and give it enough privileges to run the external-attacher. We provide one omnipotent yaml file that creates everything that's necessary, however it should be split into multiple files in production.
 
-As consequence, the attacher must be available until all PVs that refer to the CSI driver *are removed*. Even fully detached PVs have attacher's finalizer that is removed only after the PV is marked for deletion.
+```sh
+kubectl create deploy/kubernetes/deployment.yaml
+```
 
-#### Alternatives considered
+Note that the external-attacher does not scale with more replicas. Only one external-attacher is elected as leader and running. The others are waiting for the leader to die. They re-elect a new active leader in ~15 seconds after death of the old leader.
 
-Secondary cache and locks was considered to keep a map PV -> list of VolumeAttachments that use the PV. Attacher's finalizer could be removed from a PV immediately after the last VolumeAttachment was deleted. Keeping this map is either racy or requires long critical sections with complicated error recovery.
+### Command line options
 
-## Usage
+#### Required arguments that must be set
+* `--csi-address <path to CSI socket>`: This is the path to the CSI driver socket inside the pod that the external-attacher container will use to issue CSI operations (e.g. `/csi/csi.sock`).
```

Comment:
The code actually defaults this, although we don't use this default in any of our examples...

Reply:
Added default and moved to "important options" below.